pkg/scheduler/core/generic_scheduler.go

```diff
@@ -94,20 +94,20 @@ func (g *genericScheduler) Schedule(
 	}
 	klog.V(4).Infof("Feasible clusters scores: %v", clustersScore)
 
-	clusters, err := g.selectClusters(clustersScore, spec.Placement, spec)
+	candidateClusters, err := g.selectClusters(clustersScore, spec.Placement, spec)
 	if err != nil {
 		return result, fmt.Errorf("failed to select clusters: %w", err)
 	}
-	klog.V(4).Infof("Selected clusters: %v", clusters)
+	klog.V(4).Infof("Selected clusters: %v", candidateClusters)
 
-	clustersWithReplicas, err := g.assignReplicas(clusters, spec.Placement, spec)
+	clustersWithReplicas, err := g.assignReplicas(candidateClusters, feasibleClusters, spec.Placement, spec)
 	if err != nil {
 		return result, fmt.Errorf("failed to assign replicas: %w", err)
 	}
 	klog.V(4).Infof("Assigned Replicas: %v", clustersWithReplicas)
 
 	if scheduleAlgorithmOption.EnableEmptyWorkloadPropagation {
-		clustersWithReplicas = attachZeroReplicasCluster(clusters, clustersWithReplicas)
+		clustersWithReplicas = attachZeroReplicasCluster(candidateClusters, clustersWithReplicas)
```

Comment:
thanks for your work~I have some questions
- Why is the input parameter of func `attachZeroReplicasCluster` is `candidateClusters` instead of `feasibleClusters`?
- In the scenario of using spreadConstraints with maxGroup set to 2,
 if the previous scheduling result is [member1 replicas 2, member2 replicas 1], 
 feasibleClusters are [member1, member2, member3],
 candidateClusters are [member1, member3], 
 the final assign result is [member1 replicas 2, member2 replicas 1]. 
 In this case, the return result of this function is [member1 replicas 2, member2 replicas 1, member3 replicas 0]. The length exceeds maxGroup. Will it cause misunderstanding?